### PR TITLE
test(git-archive-file): add unit test

### LINKED ISSUE #1081: Test(git-archive-file): add unit test for it
## File
[git-archive-file](https://github.com/tj/git-extras/blob/master/bin/git-archive-file)

## Testcases
* If on the branch `tags*`, create a zip file which named `{dirname}.{git describe}.zip` in the root.
* If on any branch not like `tags*`
    + and the branch is not `git_extra_default_branch`, create a zip file which is named `{dirname}.{git describe --always --long}.{branch_name}.zip` in the root.
    + and the branch is `git_extra_default_branch`, create a zip file which is named `{dirname}.{git describe --always --long}.zip` in the root.
* Base above, replace the `/` or `\` with `-` when the final filename has one.
    + branch and version name maybe include the slash `/`, but cannot include the backslash `\`
    + dirname maybe include the backslash `\`, but cannot include the slash `/`

## References
* [git_extra_default_branch](https://github.com/tj/git-extras/blob/master/helper/git-extra-utility#L8)

### PR DESCRIPTION
**Pull request by vanpipy**
Close #1081

### CODE REVIEW

File: tests/test_archive_file.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import os
+
+# Testcases issue https://github.com/tj/git-extras/issues/1081
+class TestGitAbort:
```
Comment (spacewander) [inline tests/test_archive_file.py:4]: We need to use a different name

File: bin/git-archive-file [hunk 2]
```diff
@@ -26,7 +26,6 @@ fi
 
 # rename invalid chars for the file path
 FILENAME=${FILENAME//\//-}
-FILENAME=${FILENAME//\\/-}
```
Comment (spacewander) [inline bin/git-archive-file:29]: This line is different from the above

File: tests/helper.py [hunk 3]
```diff
@@ -63,6 +66,33 @@ def teardown(self):
...
+    def invoke_installed_extras_command(self, name, *params):
+        command_name = "git-" + name
+        print(f"Invoke the git-extras command - {command_name}")
+        origin_extras_command = os.path.join(GIT_EXTRAS_BIN, command_name)
+        temp_extras_command = os.path.join(self._cwd, command_name)
+        helpers = [
+                os.path.join(GIT_EXTRAS_HELPER, "git-extra-utility"),
+                os.path.join(GIT_EXTRAS_HELPER, "is-git-repo")]
+
+        if os.path.exists(temp_extras_command) == False:
```
Comment (spacewander) [inline tests/helper.py:82]:
```suggestion
        if not os.path.exists(temp_extras_command):
```

File: tests/test_archive_file.py (hunk 1 above)
Comment (vanpipy) [reply tests/test_archive_file.py]: Fixed

File: bin/git-archive-file (hunk 2 above)
Comment (vanpipy) [reply bin/git-archive-file]: Recovered

File: tests/helper.py (hunk 3 above)
Comment (vanpipy) [reply tests/helper.py]: Done

File: bin/git-archive-file (hunk 2 above)
Comment (vanpipy) [reply bin/git-archive-file]: Need more time for this case. The preset playground did not support the preset dirname.

Comment (vanpipy) [reply bin/git-archive-file]: Alright, i take a wrong direction to figure the point out. There is only one way to pass the filter truly that `$BRANCH` has the slash or backslash. I will leave this in the issue.

Comment (vanpipy) [reply bin/git-archive-file]: Done

**Review by spacewander — APPROVED** [review]
Would you rebase to the master branch before I can merge it?

**Comment by vanpipy** [comment]
Done and great, all checks passed.

File: tests/test_archive_file.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import os, pytest
+
+# Testcases issue https://github.com/tj/git-extras/issues/1081
```
Comment (spacewander) [inline tests/test_archive_file.py:3]: We can remove the link now?

Comment (vanpipy) [reply tests/test_archive_file.py]: Removed

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
Merged. Thanks!

**Pull request merged**